### ISSUE TITLE
Add params to add attributes to a success panel to ensure screen readers can read it first. (Role, aria-labelledby) (DAC usability issue)

### ISSUE THREAD

**Issue by jrbarnes9** [ONSdigital/design-system#2340]
DAC: User should be alerted to the success panel when added to a previous page to notify user.

I suggest we add param that can be set to success type panel when used in these cases, which will add attributes to ensure screen readers can read it first. (Role, aria-labelledby)

Guidance to be added to #1643 

_Originally posted by @jrbarnes9 in https://github.com/ONSdigital/design-system/discussions/49#discussioncomment-3181190_

Page 60 of EQ RAS DAC report July 2022...

-----

Success message (usability)
The success message may be difficult to locate for some screen reader users.
WCAG Reference:
Usability
Issue ID: DAC_Success_message_usability
URL: https://surveys.onsdigital.uk/surveys/todo
Page title: Surveys to complete | ONS Business Surveys
Journey: Task 1 step 22
Screenshot:
<img width="416" alt="Screenshot 2022-07-19 at 15 20 23" src="https://user-images.githubusercontent.com/43346934/179773484-a044ee86-6206-4584-8869-f6adcc0c38b3.png">

Locating the success message ‘We have updated your telephone number’ may be a challenge for some blind users. This is because it is not introduced by a heading nor is there an alert when the page loads.
This is generally a good way of relaying success message when a new page has loaded.
Current code ref(s): #success-id > div <div class="ons-panel__body"> <p class="ons-u-fs-r" id="flashed-message-1">We have updated your telephone number</p> </div>

Screen reader comments 1: “While testing with JAWS, having updated my telephone number and pressed ‘Continue’, I was very disorientated to find that rather than being returned to the ‘Account details’ page, the ‘Surveys to complete’ page was shown, with my focus on the ‘My account’ link. In addition, it took me some time to locate the message informing me that my ‘phone number had been updated, as this was not automatically spoken, did not receive focus, and was not introduced by a heading. To make the ‘Update contact details’ feature easier for users of adaptive technology to interact with, I would suggest that consideration be given to re-displaying the ‘Account details’ page after changes have been saved, with focus being placed on a confirmation message, perhaps marked up as a heading. This text could also be automatically spoken as a status message. The issue is present with JAWS, NVDA, VoiceOver, TalkBack.”

Screen reader comments 2: “Having activated the ‘Send message’ button, I was returned to the ‘Surveys to complete’ page, but JAWS spoke no alert or status message indicating whether my message had been sent successfully, and my focus was on the heading ‘Access the Quarterly Business Survey’. I had to move to the top of the page and explore in context in order to locate the lines: ‘Completed: Message sent. View Message’. I found this frustrating, and this will probably be the case for others browsing with a screen reader. Therefore, if possible, a suitable success or failure message should be spoken automatically, and this should also appear as static text introduced by a heading, with focus being positioned at the start of this content. The issue is present with JAWS, NVDA, VoiceOver, TalkBack.”

Examples of additional instances:
Additional instances of this issue exist on other pages throughout the website; wherever this issue occurs, they too will need to be resolved.

URL: https://surveys.onsdigital.uk/surveys/help/help-completing-this-survey/answer-survey-question/send-message
Page title: Surveys to complete | ONS Business Surveys
Journey: Task 1 step 9
Note: The same issue applies on the ‘Message sent’ message.

Solution:

You could implement a h1 to introduce the changes so that screen reader users can quickly navigate to the main heading and confirm the changes have been made.
Suggestion: 
```
<h1>Confirmation</h1> <div class="ons-panel__body"> <p class="ons-u-fs-r" id="flashed-message-1">We have updated your telephone number</p> </div>
```

Alternatively, you could implement role="alert" on the div containing the message so that the message is immediately conveyed to screen reader users without interfering with their focus when the page loads.
Example: 
```
<div role="alert" class="ons-panel__body"> <p class="ons-u-fs-r" id="flashed-message-1">We have updated your telephone number</p> </div>
```

**Comment by boxadesign** [comment]
@jrbarnes9 We already do this when the type is `error` we can just extend this to include `success`
